### ISSUE TITLE
Display Image after preprocessing (Python)

### ISSUE THREAD

**Issue by jwaherr** [keras-team/keras#14466]
How can I preview or show a sample of the dataset before and after the preprocessing 

# Preprocessing the Training set
train_datagen = ImageDataGenerator(rescale = 1./255,
                                   shear_range = 0.2,
                                   zoom_range = 0.2,
                                   horizontal_flip = True) #preprocessing_function=thresholding_preprocessing)
training_set = train_datagen.flow_from_directory('dataset/training_set',
                                                 target_size = (64, 64),
                                                 batch_size = 32,
                                                 class_mode = 'binary')

# Preprocessing the Test set
test_datagen = ImageDataGenerator(rescale = 1./255)
test_set = test_datagen.flow_from_directory('dataset/test_set',
                                            target_size = (64, 64),
                                            batch_size = 32,
                                            class_mode = 'binary')

**Comment by jwaherr** [comment]
I need to document the process of the image I used cv2, matplotlib but still not working

**Comment by superp0tat0** [comment]
After you called ImageDataGenerator.flow(). You could print the type of the object it return. Which is 
<class 'tensorflow.python.keras.preprocessing.image.NumpyArrayIterator'>

You could simply call a for loop to sample some images from it. They are contained in the tuples and easy to be extracted.
My example of MNIST, you could then use matlibplot to visualize image[0]:
```
gen = ImageDataGenerator(rotation_range=8, width_shift_range=0.08, shear_range=0.3,
                         height_shift_range=0.08, zoom_range=0.08)
train_generator = gen.flow(X_train, Y_train, batch_size=128)

for image in train_generator:
  print(type(image))
  print(type(image[0]))
  print(image[0].shape)
  break

```
Got
```
<class 'tuple'>
<class 'numpy.ndarray'>
(128, 28, 28, 1)
```

**Comment by sushreebarsa** [comment]
@jwaherr Could you please try as per the [comment](https://github.com/keras-team/keras/issues/14466#issuecomment-796425752) above and let us know if this is still an issue ? Thanks!

**Comment by jwaherr** [comment]
Yes it worked, Thanks!

**Comment by jwaherr** [comment]
Yes, it worked. Thanks!
